### DISCUSSION TITLE
"Uncaught Exception" with --pptx and --pdf

### DISCUSSION THREAD

**Discussion by boldandbusted** [marp-team/marp#247] [Q&A]
Howdy. I'm encountering an error from the CLI, and I'm not sure where to start to debug this. Note that I'm running this from a 'headless' Ubuntu on WSL2. (I tried to export PPT via the embedded web server, but get an "Internal server error" with no other details in Chrome).

```
$ npx @marp-team/marp-cli@latest --pptx ./Presentation-Tasks.md
[  INFO ] Converting 1 markdown...
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "#<ErrorEvent>".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

Any help appreciated. Happy to provide more details, or run with debugging options. Thanks for this awesome tool! :)

**Comment by yhatt** [comment]
You might find out the real cause of the error by adding `DEBUG` env as `DEBUG=*`.

```
$ DEBUG=* npx @marp-team/marp-cli@latest --pptx ./Presentation-Tasks.md
```

> There is no sure way to auto-detect whether the current distribution is on WSL2 for now. So Marp will try the approach same as WSL1 but would fail immidiately.
>
> _-- #113_

In common, you have to set the path to Chrome by `CHROME_PATH` env when using Marp CLI over WSL 2. See details in #113 from frequently asked section.

```
$ CHROME_PATH=$(which google-chrome) npx @marp-team/marp-cli@latest --pptx ./Presentation-Tasks.md
```

**Comment by boldandbusted** [comment]
Fantastic tips. I'll try that. Sorry for the delay - too many notifications from GitHub! :D

**Reply by yhatt** [reply]
:+1:

The latest Marp CLI has an improved browser detection in WSL2 (https://github.com/marp-team/marp-cli/pull/410). `CHROME_PATH` is still helpful to specify the browser, but your problem might be resolved just by updating CLI in this case.
